frontend/server/src/main/java/org/pytorch/serve/device/utils/XpuUtil.java

```diff
@@ -0,0 +1,90 @@
+package org.pytorch.serve.device.utils;
```

Comment:
This seems like a big refactor, this restructure should be clearly documented in the PR description.

Reply:
Update, just saw the class diagram in the PR description. Looks good!